### DISCUSSION TITLE
Speed

### DISCUSSION THREAD

**Discussion by mx-uk** [Vedza/NitroSniperGo#95] [Q&A]
The older versions with less code are faster than the newer updated versions right?

**Comment by Vedza** [comment]
It might be a little slower, but I got messages from users who still got Nitro with it, so it seems it's fast enough.
But If you don't want any of these features added lately and want the faster it can be, you can use `2.0`

**Reply by mx-uk** [reply]
I sniped 6 months with it already but I am just wondering about speed. Do you know anything about StealthSniper, it's coded in  C++ and claims to be the fastest and all I want is speed really.

**Reply by Vedza** [reply]
It's really hard to tell which language is the faster and which sniper is the fastest as it also depends on the code of it and of the libraries used.
I did my tests between a few languages and found out Go was the best overall for requests and regular expressions speed (to detect gift links) but also to produce optimal code pretty easily (which would be "harder" in C++).
Then I did my best to code it the optimal way in Go.
And if you already got 6 months, it probably means that my sniper is at least fast enough.

**Reply by mx-uk** [reply]
I think the 6 months was luck to be honest i have no idea how it happened as I was asleep. But yeah from the bit of research I have done apparently C++ can be faster than go if it's made correctly. Also the reason I care about speed is because I am in alot of servers where people already use nitro snipers and they are quite fast so this is why speed is key for me at least haha

**Reply by Vedza** [reply]
Yeah I understand, nitro sniping is about the right sniper but also a lot of luck honestly. I would say that you shouldn't care as you've already been the fastest bot 6 times, but you can still try others snipers and see how it goes.
I would also add that this version is open source, so everyone can see there isn't any bad stuff in it and also check how bad or not is the code, which is not the case of the one you're talking about.

**Reply by mx-uk** [reply]
Yeah I understand about open source being important I know StealthSniper is good but I just don't know how good. Do you also know if hosting matters like does that make it faster because I host this sniper on a decent vps but i have no idea if it affects speed

**Reply by Vedza** [reply]
Yeah it will affect the speed of the sniper by its fast internet speed.

**Reply by mx-uk** [reply]
Does cpu speed matter at all of is it just internet speed

**Reply by Vedza** [reply]
As long as you have a decent one it's good. But of course if you have a really bad one it will affect a little bit the sniper but not as much as internet speed
